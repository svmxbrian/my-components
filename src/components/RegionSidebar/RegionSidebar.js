import React from "react";
import PropTypes from "prop-types";
import "./RegionSidebar.scss";

const closedClass = "sidebar--closed";
const openClass = "sidebar--open";

const propTypes = {
  isOpen: PropTypes.bool,
  fixed: PropTypes.bool,
  children: PropTypes.node
};
const defaultProps = {
  isOpen: false
};

class RegionSidebar extends React.PureComponent {
  state = {
    isOpen: this.props.isOpen
  };

  static defaultProps = defaultProps;
  static propTypes = propTypes;

  static getOpenClass() {
    return openClass;
  }

  static getClosedClass() {
    return closedClass;
  }

  render() {
    return (
      <div className="region-sidebar">
        <div className={this.state.isOpen ? openClass : closedClass}>
          {this.props.children}
        </div>
        <div
          className={
            this.state.isOpen ? "spacer click-to-close" : "spacer click-to-open"
          }
          onClick={() =>
            this.setState({
              isOpen: !this.state.isOpen
            })}
          />
      </div>
    );
  }
}

export default RegionSidebar;
