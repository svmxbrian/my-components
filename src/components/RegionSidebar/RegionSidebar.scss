.region-sidebar {
  display: flex;
  flex-direction: row;
  .sidebar--open {
    width: 640px;
  }

  .sidebar--closed {
    width: 320px;
  }

  .spacer {
    box-sizing: border-box;
    height: 100%;
    width: 2px;
    border: 2px solid #b6c3cc;

    &.click-to-open {
      cursor: e-resize;
    }

    &.click-to-close {
      cursor: w-resize;
    }
  }
}
