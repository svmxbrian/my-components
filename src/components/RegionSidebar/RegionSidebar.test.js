import React from 'react'
import { shallow, mount } from 'enzyme'
import RegionSidebar, { openClass, closedClass } from './RegionSidebar'

describe('RegionSidebar', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <RegionSidebar>
        <div />
      </RegionSidebar>
    )
    expect(wrapper).toBeDefined()
  })

  it(`should default to having closed sidebar`, () => {
    const sidebar = mount(
      <RegionSidebar>
        <div />
      </RegionSidebar>
    )
    expect(sidebar.find(`.${closedClass}`).length).toEqual(1)
    expect(sidebar.find(`.${openClass}`).length).toEqual(0)
  })

  it(`should have ${closedClass} class appended at closed state`, () => {
    const sidebar = mount(
      <RegionSidebar isOpen={false}>
        <div />
      </RegionSidebar>
    )
    expect(sidebar.find(`.${closedClass}`).length).toEqual(1)
    expect(sidebar.find(`.${openClass}`).length).toEqual(0)
  })

  it(`should have ${openClass} class appended at open state`, () => {
    const sidebar = mount(
      <RegionSidebar isOpen={true}>
        <div />
      </RegionSidebar>
    )
    expect(sidebar.find(`.${closedClass}`).length).toEqual(0)
    expect(sidebar.find(`.${openClass}`).length).toEqual(1)
  })

  it(`should change classes at state change for isOpen`, () => {
    const sidebar = mount(
      <RegionSidebar isOpen={false}>
        <div />
      </RegionSidebar>
    )
    expect(sidebar.find(`.${closedClass}`).length).toEqual(1)
    expect(sidebar.find(`.${openClass}`).length).toEqual(0)

    sidebar.setState({
      isOpen: true
    })

    expect(sidebar.find(`.${closedClass}`).length).toEqual(0)
    expect(sidebar.find(`.${openClass}`).length).toEqual(1)

    sidebar.setState({
      isOpen: false
    })

    expect(sidebar.find(`.${closedClass}`).length).toEqual(1)
    expect(sidebar.find(`.${openClass}`).length).toEqual(0)
  })
})
