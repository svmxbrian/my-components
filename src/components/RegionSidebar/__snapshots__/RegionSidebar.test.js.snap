// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionSidebar should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="region-sidebar"
>
    <div
        className="sidebar--closed"
    >
        <div />
    </div>
    <div
        className="spacer click-to-open"
        onClick={[Function]}
    />
</div>,
  "nodes": Array [
    <div
      className="region-sidebar"
>
      <div
            className="sidebar--closed"
      >
            <div />
      </div>
      <div
            className="spacer click-to-open"
            onClick={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <RegionSidebar
        isOpen={false}
>
        <div />
</RegionSidebar>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RegionSidebar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "isOpen": false,
        },
        "refs": Object {},
        "state": Object {
          "isOpen": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="region-sidebar"
>
          <div
                    className="sidebar--closed"
          >
                    <div />
          </div>
          <div
                    className="spacer click-to-open"
                    onClick={[Function]}
          />
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="region-sidebar"
>
          <div
                    className="sidebar--closed"
          >
                    <div />
          </div>
          <div
                    className="spacer click-to-open"
                    onClick={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RegionSidebar
    isOpen={false}
>
    <div />
</RegionSidebar>,
}
`;
